### PR TITLE
[Task] Add Basic Error Handling

### PR DESCRIPTION
**Pull request by jpeng-ms**
## Purpose
This PR added the ability to handle some basic errors from call sdk. Basic errors including:

1. invalid token
2. chat thread deleted
3. other errors when initiating chat client (network issues, etc.)

for public preview, we would only be sending errors to Contoso. For GA+, we would be having UI for those error scenarios. 
## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Pull request checklist

This PR has considered:
- [ ] Color Theming
- [x] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] View Data Injection
- [ ] Demo App UI/UX update

## How to Test
1. join the call with invalid token
2. join the call with no internet 

## Other Information

1. no internet
image
3. invalid token
image

### CODE REVIEW

**Review by Leah-Xia-Microsoft — DISMISSED** [review]

**Review by vhuseinova-msft — DISMISSED** [review]

**Review by palatter — DISMISSED** [review]
So this is a catch-all for the errors mentioned above? Are bubbling up the message?

**Comment by palatter** [comment]
As discussed on a call. The demo app will handle the error passed up and the error object contains the error message.

**Comment by palatter** [comment]
Looks good!

**Review by palatter — APPROVED** [review]
Thanks for adding the alert in the demo app, looks great!

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Views/CredentialsView.swift [hunk 1]
```diff
@@ -278,4 +280,12 @@ extension CredentialsView {
             throw DemoError.invalidToken
         }
     }
+
+    private func showError(error: ChatCompositeError) {
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
+            self.alertTitle = "Error - \(error.code)"
```
Comment (kevinyulu) [inline AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Views/CredentialsView.swift:286]: in SwiftUIDemoView, we've got a similar Alert with "Error" as title and errorMessage as message. We might consider following that error UI or make both consistent
```
            Alert(
                title: Text("Error"),
                message: Text(errorMessage),
                dismissButton:
                        .default(Text("Dismiss"), action: {
                    isErrorDisplayed = false
                })) 
```

**Review by kevinyulu — APPROVED** [review]

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Views/CredentialsView.swift (hunk 1 above)
Comment (jpeng-ms) [reply AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Views/CredentialsView.swift:286]: thanks Kevin, but I thought we are using this at line 53?

**Review by Leah-Xia-Microsoft — APPROVED** [review]

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Views/CredentialsView.swift (hunk 1 above)
Comment (jpeng-ms) [reply AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Views/CredentialsView.swift:286]: hey Kevin, let me know if I missed anything

**Pull request merged**